### PR TITLE
Make metadata-extraction.ipynb work

### PR DESCRIPTION
**Pull request by eric-anderson**
* Use materialize to save the work of re-partitioning which is >50% of notebook runtime.
* Use materialize to avoid re-computation from show/take/write
* Fix readers to use the override_num_blocks parameter instead of parallelism; that's the recommended parameter from ray, and using parallelism created confusion with the parallelism setting for ray execution
* Remove parallelism setting from the notebook. It was causing hangs.

### CODE REVIEW

File: lib/sycamore/sycamore/connectors/file/file_scan.py [hunk 1]
```diff
@@ -76,13 +76,15 @@ def __init__(
         paths: Union[str, list[str]],
         *,
         filesystem: Optional[FileSystem] = None,
-        parallelism: Optional[int] = None,
+        parallelism: Optional[str] = None,
+        override_num_blocks: Optional[int] = None,
```
Comment (bsowell) [inline lib/sycamore/sycamore/connectors/file/file_scan.py:80]: I almost wonder if we want to expose this at all. I get that it makes sense as an alternative to parallelism, but it is kind of ray specific and probably hard for people to use. Should we just leave it in `resources_args`?

**Review by bsowell — APPROVED** [review]
Code looks fine. One design question that I will leave to you.

File: lib/sycamore/sycamore/connectors/file/file_scan.py (hunk 1 above)
Comment (jonfritz) [reply lib/sycamore/sycamore/connectors/file/file_scan.py:80]: +1

**Review by jonfritz — APPROVED** [review]

File: lib/sycamore/sycamore/connectors/file/file_scan.py (hunk 1 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/connectors/file/file_scan.py:80]: We probably don't want to expose it, but at this point we have. I added a comment to consider removing it and updated the documentation to be more clear people probably don't want this.

**Pull request merged**